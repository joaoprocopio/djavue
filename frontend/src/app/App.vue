<template>
  <AppLayout>
    <RouterView />
  </AppLayout>
</template>

<script setup lang="ts">
  import AppLayout from "./AppLayout.vue"
</script>
