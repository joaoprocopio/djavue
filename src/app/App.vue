<template>
  <component :is="$route?.meta?.layout?.name ?? DefaultLayoutName">
    <RouterView />
  </component>
</template>

<script setup>
  import { DefaultLayoutName } from "~/constants"
</script>
