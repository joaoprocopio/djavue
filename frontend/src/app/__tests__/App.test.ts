import { createPinia, setActivePinia } from "pinia"
import { mount, type VueWrapper } from "@vue/test-utils"

import App from "../App.vue"

let wrapper: VueWrapper
const render = () => mount(App)

beforeAll(() => {
  global.ResizeObserver = class ResizeObserver {
    observe() {}
    unobserve() {}
    disconnect() {}
  }
})

describe("App", () => {
  beforeEach(() => {
    setActivePinia(createPinia())

    wrapper = render()
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it("Theme renders - light", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
