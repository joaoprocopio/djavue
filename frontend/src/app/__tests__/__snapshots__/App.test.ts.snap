// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Theme renders - light 1`] = `
"<div class=\\"v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr\\" style=\\"z-index: 1000;\\">
  <div class=\\"v-application__wrap\\">
    <header class=\\"v-toolbar v-toolbar--density-default elevation-0 v-theme--light v-locale--is-ltr v-app-bar\\" style=\\"top: 0px; z-index: 1006; transform: translateY(0%); position: fixed; transition: none; left: 0px; width: calc(100% - 0px - 0px);\\">
      <!---->
      <div class=\\"v-toolbar__content\\" style=\\"height: 64px;\\">
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
      <transition-stub name=\\"expand-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <!---->
      </transition-stub>
    </header>
    <main class=\\"v-main\\" style=\\"--v-layout-left: 0px; --v-layout-right: 0px; --v-layout-top: 64px; --v-layout-bottom: 56px; transition: none;\\">
      <!---->
    </main>
    <header class=\\"v-bottom-navigation v-bottom-navigation--active v-theme--light v-bottom-navigation--density-default elevation-0\\" style=\\"bottom: 0px; z-index: 1004; transform: translateY(0%); position: fixed; height: 56px; transition: none; left: 0px; width: calc(100% - 0px - 0px);\\">
      <!---->
    </header>
  </div>
</div>"
`;
